core/src/Instance/B2CAuthority.cs

```diff
@@ -99,11 +103,7 @@ private bool InTrustedHostList(string host)
 
         protected override string GetDefaultOpenIdConfigurationEndpoint()
         {
-            Uri b2cAuthority = new Uri(CanonicalAuthority);
-
-            // To support host we would need to make the call to the tenant discovery end point, 
-            // which is not supported for b2clogin.com
-            string authorityUri = string.Format(CultureInfo.InvariantCulture, MicrosoftOnline + b2cAuthority.AbsolutePath + OpenIdConfigurationEndpoint);
+            string authorityUri = string.Format(CultureInfo.InvariantCulture, new Uri(CanonicalAuthority) + OpenIdConfigurationEndpoint);
```

Comment:
>ew Uri(CanonicalAuthority) [](start = 79, length = 26)

IMO, you should be more specific on which value you are using from the uri object instead of relying on .ToString()

Reply:
not sure if this is what you mean. 

---
In reply to: [234387424](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1405#discussion_r234387424) [](ancestors = 234387424)

Reply:
great, you still did the work what I was asking for :)

---
In reply to: [234388886](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1405#discussion_r234388886) [](ancestors = 234388886,234387424)

Reply:
#ItsAMiracle

---
In reply to: [234388977](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1405#discussion_r234388977) [](ancestors = 234388977,234388886,234387424)